pyogrio/raw.py

```diff
@@ -421,13 +418,13 @@ def write(
                 if not isinstance(v, str):
                     raise ValueError(f"metadata value {v} must be a string")
 
-    if promote_to_multi is None:
+    if geometry_type is not None and promote_to_multi is None:
         promote_to_multi = (
             geometry_type.startswith("Multi")
             and driver in DRIVERS_NO_MIXED_SINGLE_MULTI
         )
 
-    if crs is None:
+    if geometry_type is not None and crs is None:
```

Comment:
```suggestion
    if geometry is not None and crs is None:
```

Reply:
If `geometry_type` is specified, but `geometry=None` at the moment the geometry column is created but with all geometry values None. In this case I think the crs should still be specified, so I don't think this should be changed?

Reply:
If geometry is `None` there is no geometry column, and thus not filled with `None` values.  I was intending the change to be based on whether or not there is a `geometry` column; if there isn't then updating geometry type or needing a CRS are not relevant.  I'm not sure it is useful to allow the user to specify `geometry_type` but not `geometry`; that seems more like an error.  Or put differently, the presence of a geometry column signals that there should also be a CRS.

Reply:
As you wish, if someone wants a geometry column with all None values they can also still use the `np.repeat(None,...)` trick.

So, I'll add an error if geometry_type is not None and geometry is None?

Reply:
I don't know that we need to specifically raise an error in that case; it should be uncommon.  I think it is more that we'll simply ignore `geometry_type` if there is no geometry, which seems OK.

Reply:
Ok, it doesn't matter to me, they are indeed all uncommon cases that aren't really interesting, so as long as it they don't give weird errors or side effects it's all fine for me.

So I'll just overrule `geometry_type` and set it to `None` if `geometry` is `None`.